test/common/http/http1/codec_impl_test.cc

```diff
@@ -69,6 +115,76 @@ TEST_F(Http1ServerConnectionImplTest, Http10) {
 }
 
+TEST_F(Http1ServerConnectionImplTest, Http10AbsoluteNoOp) {
+  TestHeaderMapImpl expected_headers{{":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET / HTTP/1.0\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http10, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http10Absolute) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/foobar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/foobar HTTP/1.0\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http10, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath1) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath2) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsoluteEnabledNoOp) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "bah"}, {":path", "/foo/bar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET /foo/bar HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11InvalidRequest) {
+  Buffer::OwnedImpl buffer("GET www.somewhere.com HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathNoSlash) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com"}, {":path", "/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathBad) {
+  Buffer::OwnedImpl buffer("GET * HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePortTooLarge) {
+  Buffer::OwnedImpl buffer("GET http://foobar.com:1000000 HTTP/1.1\r\nHost: bah\r\n\r\n");
+  Expect400(Protocol::Http11, true, buffer);
+}
+
+TEST_F(Http1ServerConnectionImplTest, Http11RelativeOnly) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "bah"}, {":path", "http://www.somewhere.com/"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer("GET http://www.somewhere.com/ HTTP/1.1\r\nHost: bah\r\n\r\n");
+  ExpectHeadersTest(Protocol::Http11, false, buffer, expected_headers);
+}
```

Comment:
> which when passed on were not historically handled correctly by the next hop

Currently Envoy blocks absolute URLs, and with this change, will crack an absolute URL and make it relative. So I think it's probably fine to do always?

> but I don't know how @mattklein123 feels about Envoy blocking shady-but-not-illegal traffic.

Fine with me, but might want config flag for something like this?

> I'd say leave it in for now, and I'll poke around with our security folk to see if there's a good reason to default it.

SGTM. We can always default it to true later.